usr/share/rear/lib/global-functions.sh

```diff
@@ -558,65 +619,135 @@ mount_url() {
 
-    AddExitTask "umount -f $v '$mountpoint' >&2"
+    AddExitTask "perform_umount_url '$url' '$mountpoint' y"
     return 0
 }
 
-### Unmount url $1 at mountpoint $2
-umount_url() {
+function remove_temporary_mountpoint() {
+    if test -d "$1" ; then
+        rmdir $v "$1"
+    fi
+}
+
+### Unmount url $1 at mountpoint $2, perform mountpoint cleanup and exit task + error handling
+function umount_url() {
     local url=$1
     local mountpoint=$2
+    local scheme
 
-    case $(url_scheme $url) in
-        (null|tape|file|rsync|fish|ftp|ftps|hftp|http|https|sftp)
-            ### Don't need to umount anything for these
+    scheme=$( url_scheme $url )
+
+    # The cases where we return 0 are those that do not need umount and also do not need ExitTask handling.
+    # They thus need to be kept in sync with mount_url() so that RemoveExitTasks is used
+    # iff AddExitTask was used in mount_url().
+
+    if ! scheme_supports_filesystem $scheme ; then
+        ### Stuff like null|tape|rsync|fish|ftp|ftps|hftp|http|https|sftp
+        ### Don't need to umount anything for these.
+        ### file: supports filesystem access, but is not mounted and unmounted,
+        ### so it has to be handled specially below.
+        ### Similarly for iso: which gets mounted and unmounted only during recovery.
+        return 0
+    fi
+
+    case $scheme in
+        (file)
             return 0
             ;;
         (iso)
             if [[ "$WORKFLOW" != "recover" ]]; then
                 return 0
             fi
             ;;
-	    (sshfs)
-	        umount_cmd="fusermount -u $mountpoint"
-	    ;;
-	    (davfs)
-	        umount_cmd="umount $mountpoint"
-            # Wait for 3 sek. then remove the cache-dir /var/cache/davfs
-            sleep 30
-            # ToDo: put in here the cache-dir from /etc/davfs2/davfs.conf
-            # and delete only the just used cache
-            #rm -rf /var/cache/davfs2/*<mountpoint-hash>*
-            rm -rf /var/cache/davfs2/*outputfs*
-
-	    ;;
-        (var)
-            local var=$(url_host $url)
-            umount_cmd="${!var} $mountpoint"
+        (*)
+            # Schemes that actually need nontrivial umount are handled below.
+            # We do not handle them in the default branch because in the case of iso:
+            # it depends on the current workflow whether umount is needed or not.
+            :
+    esac
 
-            Log "Unmounting with '$umount_cmd'"
-            $umount_cmd
-            StopIfError "Unmounting failed."
+    # umount_url() is a wrapper that takes care of exit tasks and error handling and mountpoint cleanup.
+    # Therefore it also determines if exit task and mountpoint handling is required and returns early if not.
+    # The actual umount job is performed inside perform_umount_url().
+    # We do not request lazy umount here because we want umount errors to be reliably reported.
+    perform_umount_url $url $mountpoint || Error "Unmounting '$mountpoint' failed."
```

Comment:
I assume `perform_umount_url` is also called during "rear recover"
after the backup was restored but then we do not want to error out
so I wonder if here perhaps only a `LogPrintError` is sufficient?
Or do you like to keep the `Error` here and let me try to solve
the "error handling after the backup was restored" problem
in a more generic way for example as I described in
https://github.com/rear/rear/issues/2611#issuecomment-851456423

Reply:
yes, I am aware of the problem, but it has existed before and solving it is outside the scope of this PR.

Reply:
Perfectly fine with me! Leave it as is.